.pre-commit-config.yaml

```diff
@@ -10,7 +10,7 @@ repos:
     hooks:
       - id: check-bash-syntax
   - repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v0.910
+    rev: v1.16.1
```

Comment:
My understanding is that we are currently installing the latest version of `mypy` when executing the various Python 3.x tests using `nox` (e.g. see https://github.com/elastic/ecs-logging-python/blob/main/noxfile.py#L54) hence there was a discrepancy here when running precommit tests, that was surfacing non-existent errors.